bundles/org.eclipse.equinox.p2.metadata/src/org/eclipse/equinox/internal/p2/metadata/ResolvedInstallableUnit.java

```diff
@@ -240,37 +206,24 @@ public boolean satisfies(IRequirement candidate) {
 
 	@Override
 	public Object getMember(String memberName) {
-		if (MEMBER_FRAGMENTS == memberName)
-			return fragments;
-		if (MEMBER_ORIGINAL == memberName)
-			return original;
-		if (InstallableUnit.MEMBER_PROVIDED_CAPABILITIES == memberName)
-			return getProvidedCapabilities();
-		if (InstallableUnit.MEMBER_ID == memberName)
-			return getId();
-		if (InstallableUnit.MEMBER_VERSION == memberName)
-			return getVersion();
-		if (InstallableUnit.MEMBER_PROPERTIES == memberName)
-			return getProperties();
-		if (InstallableUnit.MEMBER_FILTER == memberName)
-			return getFilter();
-		if (InstallableUnit.MEMBER_ARTIFACTS == memberName)
-			return getArtifacts();
-		if (InstallableUnit.MEMBER_REQUIREMENTS == memberName)
-			return getRequirements();
-		if (InstallableUnit.MEMBER_LICENSES == memberName)
-			return getLicenses();
-		if (InstallableUnit.MEMBER_COPYRIGHT == memberName)
-			return getCopyright();
-		if (InstallableUnit.MEMBER_TOUCHPOINT_DATA == memberName)
-			return getTouchpointData();
-		if (InstallableUnit.MEMBER_TOUCHPOINT_TYPE == memberName)
-			return getTouchpointType();
-		if (InstallableUnit.MEMBER_UPDATE_DESCRIPTOR == memberName)
-			return getUpdateDescriptor();
-		if (InstallableUnit.MEMBER_SINGLETON == memberName)
-			return Boolean.valueOf(isSingleton());
-		throw new IllegalArgumentException("No such member: " + memberName); //$NON-NLS-1$
+		return switch (memberName) {
+		case MEMBER_FRAGMENTS -> fragments;
```

Comment:
Can anybody tell if it is an issue if this returns a List instead of an array?

Reply:
@merks or @mickaelistria do you have an idea?
I followed the callers a bit and got the impression that arrays are not considered as expressions at all, so could this be a hidden fix for a problem?
At least the `ResolvedIUInPCR`, `OperationGenerationTest` and `EngineTest` do not cover this code.

Reply:
Sorry, no I have no clue about this.

Reply:
I don't know much about how this class is used, so I don't have solid arguments here.
But my first impression is that it could be better to stick to arrays to:
* prevent regressions in potentially unknown consumers that are casting to array
* allow some cast the returned Object and get some stronger typing checked (ie `o instanceof String[]` works while `o instanceof List<String>` doesn't).

Reply:
Agree.
Changed the code to `fragments.toArray(IInstallableUnitFragment[]::new)`. Since this is probably not called often that's not a concern.